core/node/node_framework/src/implementations/layers/base_token_ratio_persister.rs

```diff
@@ -41,7 +48,18 @@ impl WiringLayer for BaseTokenRatioPersisterLayer {
         let master_pool_resource = context.get_resource::<PoolResource<MasterPool>>()?;
         let master_pool = master_pool_resource.get().await?;
 
-        let persister = BaseTokenRatioPersister::new(master_pool, self.config);
+        let price_api_client = context.get_resource_or_default::<PriceAPIClientResource>();
```

Comment:
Is it correct to run persister without an API client? I think if you run persister but didn't provide a client resource for it, it should be an error; otherwise it can lead to confusing problems in production (e.g. persister runs, but it persists incorrect rates). I don't think fallback to default is a correct choice here.

Reply:
I see what you're saying. My reasoning was to have the default for the i tests, as I didn't want to insert the CG resource there. WDYT? I'd love a suggestion here!

Reply:
You may alter configuration of the node for the integration tests.
I think a problem you may have here is that you miss an option to specify the config provider (e.g. `pub source: String`, and then in the node builder based on this string you can choose between e.g. `coingecko` and `mock`)